src/AutoEncoderModel.cpp

```diff
@@ -124,6 +124,15 @@ AutoEncoderModel::AutoEncoderModel() :
     else {
         printf("No %s found. Initializing new flow decoder model.\n", flowDecoderFilename); // TODO logging
     }
+
+    // Create output directories if they do not exist
+    const std::string modelsRoot{"models"};
+    if (not fs::exists(modelsRoot)) {
+        printf("Default models directory does not exist. Trying to create the directory\n");
+        if (not fs::create_directory(fs::path(modelsRoot))) {
```

Comment:
`create_directories` makes also the parent directories if necessary. I'd favor that.